### ISSUE TITLE
ValueError: invalid info for [index] for [tz], existing_value [CET] conflicts with new value [CET]

### ISSUE THREAD

**Issue by JackKelly** [nilmtk/nilmtk#360]
Bug reported by @piskopat in #355:

disaggregation fails after some instances :

``` python
 from nilmtk import HDFDataStore

eco1.set_window(start='2012-09-20 00:00:00', end=None)


mains1 = elec1.mains()
output1 = HDFDataStore('test1.h5', 'w')


co.disaggregate(mains1, output1)

Loading data for meter ElecMeterID(instance=3, building=1, dataset='ECO')     
Done loading data all meters for this chunk.
vampire_power = 52.3690032959 watts
Loading data for meter ElecMeterID(instance=3, building=1, dataset='ECO')     
Done loading data all meters for this chunk.
Estimating power demand for 'ElecMeter(instance=4, building=1, dataset='ECO', appliances=[Appliance(type='fridge', instance=1)])'
Estimating power demand for 'ElecMeter(instance=5, building=1, dataset='ECO', appliances=[Appliance(type='hair dryer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=6, building=1, dataset='ECO', appliances=[Appliance(type='coffee maker', instance=1)])'
Estimating power demand for 'ElecMeter(instance=7, building=1, dataset='ECO', appliances=[Appliance(type='kettle', instance=1)])'
Estimating power demand for 'ElecMeter(instance=8, building=1, dataset='ECO', appliances=[Appliance(type='washing machine', instance=1)])'
Estimating power demand for 'ElecMeter(instance=9, building=1, dataset='ECO', appliances=[Appliance(type='computer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=10, building=1, dataset='ECO', appliances=[Appliance(type='freezer', instance=1)])'
Loading data for meter ElecMeterID(instance=3, building=1, dataset='ECO')     
Done loading data all meters for this chunk.
Estimating power demand for 'ElecMeter(instance=4, building=1, dataset='ECO', appliances=[Appliance(type='fridge', instance=1)])'
Estimating power demand for 'ElecMeter(instance=5, building=1, dataset='ECO', appliances=[Appliance(type='hair dryer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=6, building=1, dataset='ECO', appliances=[Appliance(type='coffee maker', instance=1)])'
Estimating power demand for 'ElecMeter(instance=7, building=1, dataset='ECO', appliances=[Appliance(type='kettle', instance=1)])'
Estimating power demand for 'ElecMeter(instance=8, building=1, dataset='ECO', appliances=[Appliance(type='washing machine', instance=1)])'
Estimating power demand for 'ElecMeter(instance=9, building=1, dataset='ECO', appliances=[Appliance(type='computer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=10, building=1, dataset='ECO', appliances=[Appliance(type='freezer', instance=1)])'
Loading data for meter ElecMeterID(instance=3, building=1, dataset='ECO')     
Done loading data all meters for this chunk.
Estimating power demand for 'ElecMeter(instance=4, building=1, dataset='ECO', appliances=[Appliance(type='fridge', instance=1)])'
Estimating power demand for 'ElecMeter(instance=5, building=1, dataset='ECO', appliances=[Appliance(type='hair dryer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=6, building=1, dataset='ECO', appliances=[Appliance(type='coffee maker', instance=1)])'
Estimating power demand for 'ElecMeter(instance=7, building=1, dataset='ECO', appliances=[Appliance(type='kettle', instance=1)])'
Estimating power demand for 'ElecMeter(instance=8, building=1, dataset='ECO', appliances=[Appliance(type='washing machine', instance=1)])'
Estimating power demand for 'ElecMeter(instance=9, building=1, dataset='ECO', appliances=[Appliance(type='computer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=10, building=1, dataset='ECO', appliances=[Appliance(type='freezer', instance=1)])'
Loading data for meter ElecMeterID(instance=3, building=1, dataset='ECO')     
Done loading data all meters for this chunk.
Estimating power demand for 'ElecMeter(instance=4, building=1, dataset='ECO', appliances=[Appliance(type='fridge', instance=1)])'
Estimating power demand for 'ElecMeter(instance=5, building=1, dataset='ECO', appliances=[Appliance(type='hair dryer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=6, building=1, dataset='ECO', appliances=[Appliance(type='coffee maker', instance=1)])'
Estimating power demand for 'ElecMeter(instance=7, building=1, dataset='ECO', appliances=[Appliance(type='kettle', instance=1)])'
Estimating power demand for 'ElecMeter(instance=8, building=1, dataset='ECO', appliances=[Appliance(type='washing machine', instance=1)])'
Estimating power demand for 'ElecMeter(instance=9, building=1, dataset='ECO', appliances=[Appliance(type='computer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=10, building=1, dataset='ECO', appliances=[Appliance(type='freezer', instance=1)])'
Loading data for meter ElecMeterID(instance=3, building=1, dataset='ECO')     
Done loading data all meters for this chunk.
Estimating power demand for 'ElecMeter(instance=4, building=1, dataset='ECO', appliances=[Appliance(type='fridge', instance=1)])'
Estimating power demand for 'ElecMeter(instance=5, building=1, dataset='ECO', appliances=[Appliance(type='hair dryer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=6, building=1, dataset='ECO', appliances=[Appliance(type='coffee maker', instance=1)])'
Estimating power demand for 'ElecMeter(instance=7, building=1, dataset='ECO', appliances=[Appliance(type='kettle', instance=1)])'
Estimating power demand for 'ElecMeter(instance=8, building=1, dataset='ECO', appliances=[Appliance(type='washing machine', instance=1)])'
Estimating power demand for 'ElecMeter(instance=9, building=1, dataset='ECO', appliances=[Appliance(type='computer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=10, building=1, dataset='ECO', appliances=[Appliance(type='freezer', instance=1)])'
Loading data for meter ElecMeterID(instance=3, building=1, dataset='ECO')     
Done loading data all meters for this chunk.
Loading data for meter ElecMeterID(instance=3, building=1, dataset='ECO')     
Done loading data all meters for this chunk.
Estimating power demand for 'ElecMeter(instance=4, building=1, dataset='ECO', appliances=[Appliance(type='fridge', instance=1)])'
Estimating power demand for 'ElecMeter(instance=5, building=1, dataset='ECO', appliances=[Appliance(type='hair dryer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=6, building=1, dataset='ECO', appliances=[Appliance(type='coffee maker', instance=1)])'
Estimating power demand for 'ElecMeter(instance=7, building=1, dataset='ECO', appliances=[Appliance(type='kettle', instance=1)])'
Estimating power demand for 'ElecMeter(instance=8, building=1, dataset='ECO', appliances=[Appliance(type='washing machine', instance=1)])'
Estimating power demand for 'ElecMeter(instance=9, building=1, dataset='ECO', appliances=[Appliance(type='computer', instance=1)])'
Estimating power demand for 'ElecMeter(instance=10, building=1, dataset='ECO', appliances=[Appliance(type='freezer', instance=1)])'
Loading data for meter ElecMeterID(instance=3, building=1, dataset='ECO')     
Done loading data all meters for this chunk.
Estimating power demand for 'ElecMeter(instance=4, building=1, dataset='ECO', appliances=[Appliance(type='fridge', instance=1)])'
c:\users\t410\nilmtk\nilmtk\metergroup.py:885: UserWarning: As a quick implementation we only get Good Sections from the first meter in the meter group.  We should really return the intersection of the good sections for all meters.  This will be fixed...
  warn("As a quick implementation we only get Good Sections from"
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-7-d01adba6651b> in <module>()
      8 
      9 
---> 10 co.disaggregate(mains1, output1)
     11 

c:\users\t410\nilmtk\nilmtk\disaggregate\combinatorial_optimisation.pyc in disaggregate(self, mains, output_datastore, **load_kwargs)
    152                                         pd.DataFrame(predicted_power,
    153                                                      index=chunk.index,
--> 154                                                      columns=cols))
    155 
    156             # Copy mains data to disag output

c:\users\t410\nilmtk\nilmtk\docinherit.pyc in f(*args, **kwargs)
     44         @wraps(self.mthd, assigned=('__name__','__module__'))
     45         def f(*args, **kwargs):
---> 46             return self.mthd(obj, *args, **kwargs)
     47 
     48         return self.use_parent_doc(f, overridden)

c:\users\t410\nilmtk\nilmtk\datastore\hdfdatastore.pyc in append(self, key, value)
    154         data in the table, so be careful.
    155         """
--> 156         self.store.append(key=key, value=value)
    157         self.store.flush()
    158 

C:\Users\T410\Anaconda\lib\site-packages\pandas\io\pytables.pyc in append(self, key, value, format, append, columns, dropna, **kwargs)
    901         kwargs = self._validate_format(format, kwargs)
    902         self._write_to_group(key, value, append=append, dropna=dropna,
--> 903                              **kwargs)
    904 
    905     def append_to_multiple(self, d, value, selector, data_columns=None,

C:\Users\T410\Anaconda\lib\site-packages\pandas\io\pytables.pyc in _write_to_group(self, key, value, format, index, append, complib, encoding, **kwargs)
   1257 
   1258         # write the object
-> 1259         s.write(obj=value, append=append, complib=complib, **kwargs)
   1260 
   1261         if s.is_table and index:

C:\Users\T410\Anaconda\lib\site-packages\pandas\io\pytables.pyc in write(self, obj, axes, append, complib, complevel, fletcher32, min_itemsize, chunksize, expectedrows, dropna, **kwargs)
   3749         self.create_axes(axes=axes, obj=obj, validate=append,
   3750                          min_itemsize=min_itemsize,
-> 3751                          **kwargs)
   3752 
   3753         if not self.is_exists:

C:\Users\T410\Anaconda\lib\site-packages\pandas\io\pytables.pyc in create_axes(self, axes, obj, validate, nan_rep, data_columns, min_itemsize, **kwargs)
   3333         self.index_axes = [
   3334             index_axes_map[a].set_pos(j).update_info(self.info)
-> 3335             for j, a in enumerate(axes)
   3336         ]
   3337         j = len(self.index_axes)

C:\Users\T410\Anaconda\lib\site-packages\pandas\io\pytables.pyc in update_info(self, info)
   1595                         "invalid info for [%s] for [%s], existing_value [%s] "
   1596                         "conflicts with new value [%s]"
-> 1597                         % (self.name, key, existing_value, value))
   1598             else:
   1599                 if value is not None or existing_value is not None:

ValueError: invalid info for [index] for [tz], existing_value [CET] conflicts with new value [CET]
```

**Comment by JackKelly** [comment]
Hmm... I wonder if this is a Pandas or PyTables bug.  The error message seems to contradict itself!  

Are you using the latest versions of Pandas and PyTables?  If so, it might be worth searching the Pandas and PyTables issue queues for similar bugs (I did a very quick search on Pandas' issue queue but didn't find anything).  If you don't find anything then it would be wonderful if you could create a minimal code example which demonstrates this bug in Pandas (maybe it's something to do with daylight saving) and submit a new bug to the Pandas folks.

Of course, if you can't replicate this bug outside of NILMTK then we'll have a look at inside NILMTK.  If it comes to this then please supply a full minimal example (including training the CO algorithm) so we can make sure we're replicating the issue exactly.

**Comment by nipunreddevil** [comment]
Like @JackKelly, I too have a strong feeling that this is due to daylight savings. I guess a very simple test would be set the start and end window on the same side of daylight savings time change.

**Comment by nipunbatra** [comment]
@teptep: can you test again?

**Comment by ilia200** [comment]
Have you verified the bug?
Thanks

**Comment by ilia200** [comment]
It's a bug in pandas 0.22.
Fixed the issue in 0.23 (I have verified that by running a similar code)
For more details:
https://github.com/pandas-dev/pandas/issues/21542

Ilya

**Comment by PMeira** [comment]
Confirmed fixed. If someone finds an issue, I recommend reconverting the original data since it might fix a couple potential issues due to old bugs in Pandas and/or PyTables.
